internal/provider/features.go

```diff
@@ -157,6 +157,20 @@ func schemaFeatures(supportLegacyTestSuite bool) *pluginsdk.Schema {
 				},
 			},
 		},
+
+		"storage": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*schema.Schema{
+					"use_resource_manager": {
+						Type:     pluginsdk.TypeBool,
+						Optional: true,
+					},
+				},
+			},
+		},
```

Comment:
I see we should make these toggles have the same semantic, in this case it means all the exported feature toggles are full feature. Whilst, I'm adding this here so that we can test it in the acctest. Meanwhile I didn't mention it in the document, so it is not user facing yet. Otherwise, if we use ways like env var to control this flag, it doesn't make too much difference? Or is there any other better way?

Reply:
for now we can conditionally add this schema using [the (intentionally disabled) 3.0 toggle](https://github.com/hashicorp/terraform-provider-azurerm/blob/19462592dab9b1c82e39bc36df7c5afba3ec7f6f/internal/features/three_point_oh.go#L30) - once we're further along in the 3.0 lifecycle we'll look to make that user-configurable (as detailed here: https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/guides/3.0-overview) however I think this is fine to feature-gate for now?